<?xml version="1.0"?>
<sdf version="1.5">
<model name="diffcart">
    <static>false</static>

    <link name="base">
        <pose>0 0 0.1 0 0 0</pose>
        <collision name="base_collision">
            <geometry>
                <box>
                    <size>0.4 0.2 0.1</size>
                </box>
            </geometry>
        </collision>
        <visual name="base_visual">
            <geometry>
                <box>
                    <size>0.4 0.2 0.1</size>
                </box>
            </geometry>
        </visual>

        <collision name="base_ball_collision">
            <pose>-0.15 0 -0.05 0 0 0</pose>
            <geometry>
                <sphere>
                    <radius>0.05</radius>
                </sphere>
            </geometry>
            <surface>
                <friction>
                    <ode>
                        <mu>0</mu>
                        <mu2>0</mu2>
                        <slip1>1.0</slip1>
                        <slip2>1.0</slip2>
                    </ode>
                </friction>
            </surface>
        </collision>

        <visual name="base_ball_visual">
            <pose>-0.15 0 -0.05 0 0 0</pose>
            <geometry>
                <sphere>
                    <radius>0.05</radius>
                </sphere>
            </geometry>
        </visual>
    </link>

    <link name="left_wheel">
        <pose>0.1 0.13 0.1 1.5707 0 0</pose>
        <collision name="left_wheel_collision">
            <geometry>
                <cylinder>
                    <radius>0.1</radius>
                    <length>0.05</length>
                </cylinder>
            </geometry> 
        </collision>
        <visual name="left_wheeel_visual">
            <geometry>
                <cylinder>
                    <radius>0.1</radius>
                    <length>0.05</length>
                </cylinder>
            </geometry>
        </visual>
    </link>

    <link name="right_wheel">
        <pose>0.1 -0.13 0.1 1.5707 0 0</pose>
        <collision name="right_wheel_collision">
            <geometry>
                <cylinder>
                    <radius>0.1</radius>
                    <length>0.05</length>
                </cylinder>
            </geometry> 
        </collision>
        <visual name="right_wheeel_visual">
            <geometry>
                <cylinder>
                    <radius>0.1</radius>
                    <length>0.05</length>
                </cylinder>
            </geometry>
        </visual>
    </link>

    <joint type="revolute" name="base_2_left_wheel">
        <pose>0 0 -0.03 0 0 0</pose>
        <parent>base</parent>
        <child>left_wheel</child>
        <axis>
            <xyz>0 0 1</xyz>
        </axis>
    </joint>

    <joint type="revolute" name="base_2_right_wheel">
        <pose>0 0 0.03 0 0 0</pose>
        <parent>base</parent>
        <child>right_wheel</child>
        <axis>
            <xyz>0 0 1</xyz>
        </axis>
    </joint>



</model>
</sdf>
